src/WebJobs.Extensions.DurableTask/Listener/TaskEntityShim.cs

```diff
@@ -310,6 +310,8 @@ private async Task ProcessOperationRequestAsync(RequestMessage request)
             this.context.DestructOnExit = false;
             this.context.IsCompleted = false;
 
+            Entity.SetContext(this.context);
```

Comment:
Let's add a comment here explaining what we're doing.

Reply:
Sure.